internal/provider/data_source_packer_image_iteration.go

```diff
@@ -0,0 +1,229 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	packermodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var defaultPackerTimeout = time.Minute
+
+func dataSourcePackerImageIteration() *schema.Resource {
+	return &schema.Resource{
+		Description: "The Packer Image data source iteration gets the most recent iteration (or build) of an image, given a channel.",
+		ReadContext: dataSourcePackerImageRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &defaultPackerTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"bucket": {
+				Description:      "The slug of the HCP Packer Registry image bucket to pull from.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"channel": {
+				Description:      "The channel that points to the version of the image you want.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// computed outputs
+			"organization_id": {
+				Description: "The ID of the organization this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			// Actual iteration:
+
+			"id": {
```

Comment:
Is this ID guaranteed to be unique? I ask because Consul and Vault cluster IDs are unique only within a given project, so they use this [link helper](https://github.com/hashicorp/terraform-provider-hcp/blob/main/internal/provider/link.go#L71) to create a unique URL with org and project IDs.

Also wanted to note, so far we've set a pattern of reserving the ID field as an internal field for the provider, and using a prefix for the external ID (so something like `iteration_id`).

Reply:
This iteration ID is the primary ID for iterations, yes, so it will be unique for HCP among all iterations ! 🙂. Not sure whether this is okay. I can rename it to `iteration_id`, to remain consistent, though !

Reply:
I also used the `SetID` function with the same value, not sure if they are doing the same thing or not here ?

Reply:
Actually, `SetID`, should be enough, removing this entry !